Object holders and new schema

BoogieMonster1O1 opened the pull request:
Supersedes #1 

- Use object holders
- Replace one instance of reflection with a supplier

BoogieMonster1O1 committed:
Object holders

 On branch 1.16.x-dev
 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/MonoColorSkybox.java
	new file:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Fade.java
	new file:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/HeightEntry.java
	new file:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/RGBA.java
	new file:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Textures.java
	new file:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Weather.java

BoogieMonster1O1 committed:
Refactor storage

 On branch 1.16.x-dev
 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java

BoogieMonster1O1 committed:
Remove fields

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/MonoColorSkybox.java

BoogieMonster1O1 committed:
Convert more fields into object holders

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/MonoColorSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/SquareTexturedSkybox.java

BoogieMonster1O1 committed:
Use functional interfaces instead of reflection

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/FabricSkyBoxesClient.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/SkyboxManager.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/resource/SkyboxResourceLoader.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java

AMereBagatelle reviewed (approved):
Looks good, are you finished or is there a reason this is still a draft?

BoogieMonster1O1 committed:
Serialization

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/resource/SkyboxResourceLoader.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/MonoColorSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/SquareTexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/TexturedSkybox.java

BoogieMonster1O1 commented:
> Looks good, are you finished or is there a reason this is still a draft?

I've made the new schema and added the codecs for them. 
However, I still haven't finished the decoding with the help of the codecs. I'll mark this as ready for merge once done.

BoogieMonster1O1 committed:
Codec based decoding and optionals

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/SkyboxManager.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/resource/SkyboxResourceLoader.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/JsonObjectWrapper.java

BoogieMonster1O1 committed:
Default value

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Fade.java

BoogieMonster1O1 committed:
Tests

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/FabricSkyBoxesClient.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/MonoColorSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/SquareTexturedSkybox.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/util/Utils.java
	modified:   src/main/resources/fabric.mod.json

BoogieMonster1O1 committed:
Tests

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/FabricSkyBoxesClient.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/SkyboxManager.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java
	renamed:    src/testmod/resources/assets/fabricskyboxes/sky/test_mono.json -> src/testmod/resources/assets/fabricskyboxes/sky/test_mono.json.txt
	modified:   src/testmod/resources/assets/fabricskyboxes/sky/test_textured.json
	new file:   src/testmod/resources/assets/fabricskyboxes/sky/test_texured_v2.json

BoogieMonster1O1 committed:
Start working on schema docs

 Changes to be committed:
	new file:   docs/schema_v2.md
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/FabricSkyBoxesClient.java
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/SquareTexturedSkybox.java

BoogieMonster1O1 committed:
Optional fields

 Changes to be committed:
	renamed:    docs/schema_v2.md -> docs/schema-v2.md

BoogieMonster1O1 committed:
Lenient Json and example

 Changes to be committed:
	modified:   docs/schema-v2.md
	new file:   example/sky_template_textured_v2.json
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/resource/SkyboxResourceLoader.java
	modified:   src/testmod/resources/assets/fabricskyboxes/sky/test_texured_v2.json

BoogieMonster1O1 commented:
Finished

BoogieMonster1O1 committed:
xamples

C=hanges to be committed:
modified:   docs/schema-v2.md

AMereBagatelle commented on src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/textured/TexturedSkybox.java:
```diff
@@ -1,21 +1,38 @@
+import net.minecraft.util.Identifier;
 
 public abstract class TexturedSkybox extends AbstractSkybox {
-    public float[] axis;
```
Is there a reason why this needed to change?

BoogieMonster1O1 replied:
it's now a float list instead of an array

AMereBagatelle commented on src/main/java/io/github/amerebagatelle/fabricskyboxes/FabricSkyBoxesClient.java:
```diff
@@ -29,4 +46,68 @@ public static Logger getLogger() {
         return LOGGER;
     }
+
+    public static void test() {
```
I'd like it if this could be separated to a different class.  It doesn't seem quite fitting to have in the client initializer.

AMereBagatelle reviewed (changes requested):
Looks good in general, just a few things I have questions/suggestions on.

BoogieMonster1O1 committed:
Move prelaunch entrypoint to seperate class

 Changes to be committed:
	modified:   src/main/java/io/github/amerebagatelle/fabricskyboxes/FabricSkyBoxesClient.java
	new file:   src/main/java/io/github/amerebagatelle/fabricskyboxes/Test.java
	modified:   src/main/resources/fabric.mod.json

AMereBagatelle reviewed (approved):
Looks good, thank you!
